docs/docs/how-to/performance/improving-site-performance.md

```diff
@@ -0,0 +1,203 @@
+
+#### Second, remove unneeded third-party imports from commons.js
+
+Start by auditing your `commons.js` -- the bundle that is shared by all components.
+
+The first thing to do is inspect third-party package size. Anything over 50kb, and certainly 100kb, is worth examining whether it's needed. Some common culprits include: Moment.js ([see deprecation notice](https://momentjs.com/docs/#/-project-status/)), Lodash, Material UI (see [this description](https://www.freecodecamp.org/news/gatsby-perfect-lighthouse-score/#step3droppingmaterialuifortailwindcss)), but you'll want to inspect your individual libraries.
+
+To prevent large imports from recurring, consider using a tool that displays the size of library imports you're pulling in. The [Import Cost](https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost) extension for Visual Studio Code and [BundlePhobia](https://bundlephobia.com/) are good resources.
+
+In addition, eyeball all the medium-sized packages (10-50kb). If it doesn't look like there's a good reason for that particular package to be in the commons, carefully audit your import structure
+
+One edge case: If you're [importing Redux globally](/docs/adding-redux-store/), Redux can pull in data bundles that don't seem to be related. If you're only using Redux on a page or two, you may want to check the impact by removing it entirely and running Lighthouse. 
+
+#### Third, audit your commons.js for components and data that don't need to be on every page
+
+One common challenge is inadvertently pulling in more code or data than you intend to your commons bundle.
+
+For example, let's say you have a header that imports a JSON object in order to check a key / value pair on that object. Whether that JSON object is 5kb or 50kb, you've now imported it into every page on your site!
+
+There's a couple ways to detect this:
+
+- Notice components and data that don't seem to be needed on every page. If you're using v2 of Gatsby, certain methods of importing can cause code to get bundled on pages it doesn't belong on. Try replacing indirect import statements like `import { myIcon } from './icons/index.js'` with direct imports like `import { myIcon} from './icons/my-icon.js`. 
+
+- Watch for unexpectedly large data imports. If you notice large JSON objects, and you do need the data (or some portion of it), there are a couple options.
+
+- If you only need a small portion of that data, consider getting it a different way. Split up the JSON file, [query it via GraphQL](https://www.gatsbyjs.com/plugins/gatsby-transformer-json/), or import it in `gatsby-node.js` and pass through only the subset of data you need. 
+
+- If you need the data, but not right away (perhaps it's lower in the page, or being used by an event handler), you might consider switching to asynchronously fetching it.  
```

Comment:
Are these two supposed to be nested under the bullet on line 111?

Reply:
ah, good call, yep!